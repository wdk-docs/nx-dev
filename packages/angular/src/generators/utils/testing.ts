import type { Tree } from '@nrwl/devkit';
import { wrapAngularDevkitSchematic } from '@nrwl/devkit/ngcli-adapter';
import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';
import { Linter } from '@nrwl/linter';
import { UnitTestRunner } from '../../utils/test-runners';
import libraryGenerator from '../library/library';

export async function createStorybookTestWorkspaceForLib(
  libName: string
): Promise<Tree> {
  let tree = createTreeWithEmptyWorkspace();

  const moduleGenerator = wrapAngularDevkitSchematic(
    '@schematics/angular',
    'module'
  );
  const componentGenerator = wrapAngularDevkitSchematic(
    '@schematics/angular',
    'component'
  );

  await libraryGenerator(tree, {
    name: libName,
    buildable: false,
    linter: Linter.EsLint,
    publishable: false,
    simpleModuleName: false,
    skipFormat: false,
    unitTestRunner: UnitTestRunner.Jest,
  });

  await componentGenerator(tree, {
    name: 'test-button',
    project: libName,
  });

  tree.write(
    `libs/${libName}/src/lib/test-button/test-button.component.ts`,
    `import { Component, Input } from '@angular/core';

export type ButtonStyle = 'default' | 'primary' | 'accent';

@Component({
  selector: 'proj-test-button',
  templateUrl: './test-button.component.html',
  styleUrls: ['./test-button.component.css']
})
export class TestButtonComponent {
  @Input('buttonType') type = 'button';
  @Input() style: ButtonStyle = 'default';
  @Input() age?: number;
  @Input() isOn = false;
}`
  );

  tree.write(
    `libs/${libName}/src/lib/test-button/test-button.component.html`,
    `<button [attr.type]="type" [ngClass]="style"></button>`
  );

  const modulePath = `libs/${libName}/src/lib/${libName}.module.ts`;
  tree.write(
    modulePath,
    `import * as ButtonExports from './test-button/test-button.component';
    ${tree.read(modulePath)}`
  );

  // create a module with component that gets exported in a barrel file
  await moduleGenerator(tree, {
    name: 'barrel',
    project: libName,
  });

  await componentGenerator(tree, {
    name: 'barrel-button',
    project: libName,
    path: `libs/${libName}/src/lib/barrel`,
    module: 'barrel',
  });

  tree.write(
    `libs/${libName}/src/lib/barrel/barrel-button/index.ts`,
    `export * from './barrel-button.component';`
  );

  tree.write(
    `libs/${libName}/src/lib/barrel/barrel.module.ts`,
    `import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BarrelButtonComponent } from './barrel-button';

@NgModule({
  imports: [CommonModule],
  declarations: [BarrelButtonComponent],
})
export class BarrelModule {}`
  );

  // create a module with components that get Angular exported and declared by variable
  await moduleGenerator(tree, {
    name: 'variable-declare',
    project: libName,
  });

  await componentGenerator(tree, {
    name: 'variable-declare-button',
    project: libName,
    path: `libs/${libName}/src/lib/variable-declare`,
    module: 'variable-declare',
  });

  await componentGenerator(tree, {
    name: 'variable-declare-view',
    project: libName,
    path: `libs/${libName}/src/lib/variable-declare`,
    module: 'variable-declare',
  });

  tree.write(
    `libs/${libName}/src/lib/variable-declare/variable-declare.module.ts`,
    `import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { VariableDeclareButtonComponent } from './variable-declare-button/variable-declare-button.component';
import { VariableDeclareViewComponent } from './variable-declare-view/variable-declare-view.component';

const COMPONENTS = [
  VariableDeclareButtonComponent,
  VariableDeclareViewComponent
]

@NgModule({
  imports: [CommonModule],
  declarations: COMPONENTS,
  exports: COMPONENTS
})
export class VariableDeclareModule {}`
  );

  // create a module with components that get Angular exported and declared by variable
  await moduleGenerator(tree, {
    name: 'variable-spread-declare',
    project: libName,
  });

  await componentGenerator(tree, {
    name: 'variable-spread-declare-button',
    project: libName,
    path: `libs/${libName}/src/lib/variable-spread-declare`,
    module: 'variable-spread-declare',
  });

  await componentGenerator(tree, {
    name: 'variable-spread-declare-view',
    project: libName,
    path: `libs/${libName}/src/lib/variable-spread-declare`,
    module: 'variable-spread-declare',
  });

  await componentGenerator(tree, {
    name: 'variable-spread-declare-anotherview',
    project: libName,
    path: `libs/${libName}/src/lib/variable-spread-declare`,
    module: 'variable-spread-declare',
  });

  tree.write(
    `libs/${libName}/src/lib/variable-spread-declare/variable-spread-declare.module.ts`,
    `import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { VariableSpreadDeclareButtonComponent } from './variable-spread-declare-button/variable-spread-declare-button.component';
import { VariableSpreadDeclareViewComponent } from './variable-spread-declare-view/variable-spread-declare-view.component';
import { VariableSpreadDeclareAnotherviewComponent } from './variable-spread-declare-anotherview/variable-spread-declare-anotherview.component';

const COMPONENTS = [ 
  VariableSpreadDeclareButtonComponent, 
  VariableSpreadDeclareViewComponent 
]

@NgModule({
  imports: [CommonModule],
  declarations: [...COMPONENTS, VariableSpreadDeclareAnotherviewComponent],
})
export class VariableSpreadDeclareModule {}`
  );

  // create a module where declared components are pulled from a static member of the module
  await moduleGenerator(tree, {
    name: 'static-member-declarations',
    project: libName,
  });

  await componentGenerator(tree, {
    name: 'cmp1',
    project: libName,
    path: `libs/${libName}/src/lib/static-member-declarations`,
    module: 'static-member-declarations',
  });

  await componentGenerator(tree, {
    name: 'cmp2',
    project: libName,
    path: `libs/${libName}/src/lib/static-member-declarations`,
    module: 'static-member-declarations',
  });

  tree.write(
    `libs/${libName}/src/lib/static-member-declarations/static-member-declarations.module.ts`,
    `import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { Cmp1Component } from './cmp1/cmp1.component';
import { Cmp2Component } from './cmp2/cmp2.component';

@NgModule({
  imports: [CommonModule],
  declarations: StaticMemberDeclarationsModule.COMPONENTS,
  exports: StaticMemberDeclarationsModule.COMPONENTS
})
export class StaticMemberDeclarationsModule {
  static readonly COMPONENTS = [Cmp1Component, Cmp2Component];
}`
  );

  // create another button in a nested subpath
  await moduleGenerator(tree, {
    name: 'nested',
    project: libName,
    path: `libs/${libName}/src/lib`,
  });

  await componentGenerator(tree, {
    name: 'nested-button',
    project: libName,
    module: 'nested',
    path: `libs/${libName}/src/lib/nested`,
  });

  await componentGenerator(tree, {
    name: 'test-other',
    project: libName,
  });

  return tree;
}
