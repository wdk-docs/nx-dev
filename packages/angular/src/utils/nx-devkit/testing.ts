import type { Tree } from '@nrwl/devkit';
import { names, updateJson } from '@nrwl/devkit';

export interface AppConfig {
  appName: string; // name of app
  appModule: string; // app/app.module.ts in the above sourceDir
}
export interface LibConfig {
  name: string;
  module: string;
  barrel: string;
}

var appConfig: AppConfig; // configure built in createApp()
var libConfig: LibConfig;

export function getAppConfig(): AppConfig {
  return appConfig;
}
export function getLibConfig(): LibConfig {
  return libConfig;
}

export function createApp(
  tree: Tree,
  appName: string,
  routing: boolean = true
) {
  appName = names(appName).fileName;
  // save for getAppDir() lookup by external *.spec.ts tests
  appConfig = {
    appName,
    appModule: `/apps/${appName}/src/app/app.module.ts`,
  };

  tree.write(
    appConfig.appModule,
    `
     import { NgModule } from '@angular/core';
     import { BrowserModule } from '@angular/platform-browser';
     ${routing ? "import { RouterModule } from '@angular/router'" : ''};
     import { AppComponent } from './app.component';
     @NgModule({
       imports: [BrowserModule, ${routing ? 'RouterModule.forRoot([])' : ''}],
       declarations: [AppComponent],
       bootstrap: [AppComponent]
     })
     export class AppModule {}
  `
  );
  tree.write(
    `/apps/${appName}/src/main.ts`,
    `
    import { enableProdMode } from '@angular/core';
    import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';

    import { AppModule } from './app/app.module';
    import { environment } from './environments/environment';

    if (environment.production) {
      enableProdMode();
    }

    platformBrowserDynamic()
      .bootstrapModule(AppModule)
      .catch(err => console.log(err));
  `
  );
  tree.write(
    `/apps/${appName}/tsconfig.app.json`,
    JSON.stringify({
      include: ['**/*.ts'],
    })
  );
  tree.write(
    `/apps/${appName}-e2e/tsconfig.e2e.json`,
    JSON.stringify({
      include: ['../**/*.ts'],
    })
  );
  tree.write(
    '/workspace.json',
    JSON.stringify({
      newProjectRoot: '',
      version: 1,
      projects: {
        [appName]: {
          root: `apps/${appName}`,
          sourceRoot: `apps/${appName}/src`,
          architect: {
            build: {
              options: {
                main: `apps/${appName}/src/main.ts`,
              },
            },
            serve: {
              options: {},
            },
          },
          tags: [],
        },
      },
    })
  );
}

export function createLib(tree: Tree, libName: string) {
  const { name, className, fileName, propertyName } = names(libName);

  libConfig = {
    name,
    module: `/libs/${propertyName}/src/lib/${fileName}.module.ts`,
    barrel: `/libs/${propertyName}/src/index.ts`,
  };

  tree.write(
    libConfig.module,
    `
      import { NgModule } from '@angular/core';
      import { CommonModule } from '@angular/common';
      @NgModule({
        imports: [
          CommonModule
        ],
        providers: []
      })
      export class ${className}Module { }
  `
  );
  tree.write(
    libConfig.barrel,
    `
    export * from './lib/${fileName}.module';
  `
  );
}
