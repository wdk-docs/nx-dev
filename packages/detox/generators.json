{
  "name": "Nx Detox",
  "version": "0.1",
  "extends": ["@nrwl/workspace"],
  "schematics": {
    "init": {
      "factory": "./src/generators/init/init#detoxInitSchematic",
      "schema": "./src/generators/init/schema.json",
      "description": "Initialize the @nrwl/detox plugin",
      "hidden": true
    },
    "application": {
      "factory": "./src/generators/application/application#detoxApplicationSchematic",
      "schema": "./src/generators/application/schema.json",
      "aliases": ["app"],
      "x-type": "application",
      "description": "Create a detox application"
    }
  },
  "generators": {
    "init": {
      "factory": "./src/generators/init/init#detoxInitGenerator",
      "schema": "./src/generators/init/schema.json",
      "description": "Initialize the @nrwl/detox plugin",
      "hidden": true
    },
    "application": {
      "factory": "./src/generators/application/application#detoxApplicationGenerator",
      "schema": "./src/generators/application/schema.json",
      "aliases": ["app"],
      "x-type": "application",
      "description": "Create a detox application"
    }
  }
}
