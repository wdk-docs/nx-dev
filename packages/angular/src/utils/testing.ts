import { join } from 'path';
import { SchematicTestRunner } from '@angular-devkit/schematics/testing';
import { Rule, Tree } from '@angular-devkit/schematics';

const testRunner = new SchematicTestRunner(
  '@nrwl/angular',
  join(__dirname, '../../generators.json')
);

testRunner.registerCollection(
  '@nrwl/jest',
  join(__dirname, '../../../jest/generators.json')
);

testRunner.registerCollection(
  '@nrwl/workspace',
  join(__dirname, '../../../workspace/generators.json')
);

testRunner.registerCollection(
  '@nrwl/cypress',
  join(__dirname, '../../../cypress/generators.json')
);

testRunner.registerCollection(
  '@nrwl/storybook',
  join(__dirname, '../../../storybook/generators.json')
);

const migrationTestRunner = new SchematicTestRunner(
  '@nrwl/workspace',
  join(__dirname, '../../migrations.json')
);

export function runMigration<SchemaOptions = any>(
  schematicName: string,
  options: SchemaOptions,
  tree: Tree
) {
  return migrationTestRunner
    .runSchematicAsync(schematicName, options, tree)
    .toPromise();
}

export function callRule(rule: Rule, tree: Tree) {
  return testRunner.callRule(rule, tree).toPromise();
}
