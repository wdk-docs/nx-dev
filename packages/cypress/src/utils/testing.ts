import { join } from 'path';
import { SchematicTestRunner } from '@angular-devkit/schematics/testing';
import { Tree, Rule } from '@angular-devkit/schematics';

const testRunner = new SchematicTestRunner(
  '@nrwl/cypress',
  join(__dirname, '../../generators.json')
);

const migrationRunner = new SchematicTestRunner(
  '@nrwl/cypress/migrations',
  join(__dirname, '../../migrations.json')
);

export function runSchematic(schematicName: string, options: any, tree: Tree) {
  return testRunner.runSchematicAsync(schematicName, options, tree).toPromise();
}

export function callRule(rule: Rule, tree: Tree) {
  return testRunner.callRule(rule, tree).toPromise();
}

export function runMigration(migrationName: string, options: any, tree: Tree) {
  return migrationRunner
    .runSchematicAsync(migrationName, options, tree)
    .toPromise();
}
