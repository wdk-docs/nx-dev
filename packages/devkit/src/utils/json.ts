import type { Tree } from '@nrwl/tao/src/shared/tree';
import { parseJson, serializeJson } from '@nrwl/tao/src/utils/json';
import type {
  JsonParseOptions,
  JsonSerializeOptions,
} from '@nrwl/tao/src/utils/json';

/**
 * Reads a json file, removes all comments and parses JSON.
 *
 * @param tree - file system tree
 * @param path - file path
 * @param options - Optional JSON Parse Options
 */
export function readJson<T extends object = any>(
  tree: Tree,
  path: string,
  options?: JsonParseOptions
): T {
  if (!tree.exists(path)) {
    throw new Error(`Cannot find ${path}`);
  }
  try {
    return parseJson(tree.read(path, 'utf-8'), options);
  } catch (e) {
    throw new Error(`Cannot parse ${path}: ${e.message}`);
  }
}

/**
 * Writes a JSON value to the file system tree

 * @param tree File system tree
 * @param path Path of JSON file in the Tree
 * @param value Serializable value to write
 * @param options Optional JSON Serialize Options
 */
export function writeJson<T extends object = object>(
  tree: Tree,
  path: string,
  value: T,
  options?: JsonSerializeOptions
): void {
  tree.write(path, serializeJson(value, options));
}

/**
 * Updates a JSON value to the file system tree
 *
 * @param tree File system tree
 * @param path Path of JSON file in the Tree
 * @param updater Function that maps the current value of a JSON document to a new value to be written to the document
 * @param options Optional JSON Parse and Serialize Options
 */
export function updateJson<T extends object = any, U extends object = T>(
  tree: Tree,
  path: string,
  updater: (value: T) => U,
  options?: JsonParseOptions & JsonSerializeOptions
): void {
  const updatedValue = updater(readJson(tree, path, options));
  writeJson(tree, path, updatedValue, options);
}
