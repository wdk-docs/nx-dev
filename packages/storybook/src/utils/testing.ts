import { join } from 'path';
import { externalSchematic, Rule, Tree } from '@angular-devkit/schematics';
import { SchematicTestRunner } from '@angular-devkit/schematics/testing';

import { createEmptyWorkspace } from '@nrwl/workspace/testing';

const testRunner = new SchematicTestRunner(
  '@nrwl/storybook',
  join(__dirname, '../../generators.json')
);

['angular', 'react', 'jest', 'cypress'].forEach((collection) =>
  testRunner.registerCollection(
    `@nrwl/${collection}`,
    join(__dirname, `../../../${collection}/generators.json`)
  )
);

const migrationRunner = new SchematicTestRunner(
  '@nrwl/storybook/migrations',
  join(__dirname, '../../migrations.json')
);

export function runSchematic(schematicName: string, options: any, tree: Tree) {
  return testRunner.runSchematicAsync(schematicName, options, tree).toPromise();
}

export function runExternalSchematic(
  collectionName: string,
  schematicName: string,
  options: any,
  tree: Tree
) {
  return testRunner
    .runExternalSchematicAsync(collectionName, schematicName, options, tree)
    .toPromise();
}

export function callRule(rule: Rule, tree: Tree) {
  return testRunner.callRule(rule, tree).toPromise();
}

export function runMigration(migrationName: string, options: any, tree: Tree) {
  return migrationRunner
    .runSchematicAsync(migrationName, options, tree)
    .toPromise();
}

export async function createTestUILib(
  libName: string,
  collectionName: '@nrwl/angular' | '@nrwl/react',
  options: any = {}
): Promise<Tree> {
  let appTree = Tree.empty();
  appTree = createEmptyWorkspace(appTree);
  appTree = await callRule(
    externalSchematic(collectionName, 'library', {
      name: libName,
      ...options,
    }),
    appTree
  );
  appTree = await callRule(
    externalSchematic(collectionName, 'component', {
      name: 'test-button',
      project: libName,
    }),
    appTree
  );

  if (collectionName === '@nrwl/angular') {
    updateAngularComponent(appTree);
  }
  if (collectionName === '@nrwl/react') {
    // @TODO
  }

  appTree = await callRule(
    externalSchematic(collectionName, 'component', {
      name: 'test-other',
      project: libName,
    }),
    appTree
  );

  return appTree;

  function updateAngularComponent(appTree: Tree) {
    appTree.overwrite(
      `libs/${libName}/src/lib/test-button/test-button.component.ts`,
      `
import { Component, OnInit, Input } from '@angular/core';
import { tmpdir } from 'os';
import { mkdtempSync } from 'fs';

export type ButtonStyle = 'default' | 'primary' | 'accent';

@Component({
  selector: 'proj-test-button',
  templateUrl: './test-button.component.html',
  styleUrls: ['./test-button.component.css']
})
export class TestButtonComponent implements OnInit {
  @Input('buttonType') type = 'button';
  @Input() style: ButtonStyle = 'default';
  @Input() age: number;
  @Input() isOn = false;

  constructor() { }

  ngOnInit() {
  }

}
`
    );
    appTree.overwrite(
      `libs/${libName}/src/lib/test-button/test-button.component.html`,
      `<button [attr.type]="type" [ngClass]="style"></button>`
    );
  }
}
