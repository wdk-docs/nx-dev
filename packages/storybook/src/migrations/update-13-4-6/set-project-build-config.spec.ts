import { Tree, writeJson } from '@nrwl/devkit';
import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';
import { readWorkspace } from 'packages/devkit/src/generators/project-configuration';
import setProjectBuildConfig from './set-project-build-config';
import * as defaultConfig from './test-configs/default-config.json';
import * as customNames from './test-configs/custom-names-config.json';
import * as nonAngular from './test-configs/non-angular.json';

describe('Set the projectBuildConfig option in the Storybook configuration for Angular projects', () => {
  let tree: Tree;

  describe('for all types of angular projects - non-buildable and buildable libs/apps', () => {
    beforeEach(async () => {
      tree = createTreeWithEmptyWorkspace();
    });

    it(`should set the projectBuildConfig in the Storybook config according to the type of project`, async () => {
      writeJson(tree, 'workspace.json', defaultConfig);
      await setProjectBuildConfig(tree);
      expect(readWorkspace(tree)).toMatchSnapshot();
    });

    it(`should still set the projectBuildConfig even if target names are not the default`, async () => {
      writeJson(tree, 'workspace.json', customNames);
      await setProjectBuildConfig(tree);
      expect(readWorkspace(tree)).toMatchSnapshot();
    });
  });

  describe('for non-angular projects', () => {
    beforeEach(async () => {
      tree = createTreeWithEmptyWorkspace();
      writeJson(tree, 'workspace.json', nonAngular);
    });

    it(`should not change their Storybook configuration`, async () => {
      await setProjectBuildConfig(tree);
      expect(readWorkspace(tree)).toMatchSnapshot();
    });
  });
});
