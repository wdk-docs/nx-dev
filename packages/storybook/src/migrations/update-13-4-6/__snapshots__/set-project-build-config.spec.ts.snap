// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Set the projectBuildConfig option in the Storybook configuration for Angular projects for all types of angular projects - non-buildable and buildable libs/apps should set the projectBuildConfig in the Storybook config according to the type of project 1`] = `
Object {
  "projects": Object {
    "main-app": Object {
      "prefix": "katst",
      "projectType": "application",
      "root": "apps/main-app",
      "sourceRoot": "apps/main-app/src",
      "targets": Object {
        "build": Object {
          "executor": "@angular-devkit/build-angular:browser",
          "outputs": Array [
            "{options.outputPath}",
          ],
        },
        "build-storybook": Object {
          "configurations": Object {
            "ci": Object {
              "quiet": true,
            },
          },
          "executor": "@nrwl/storybook:build",
          "options": Object {
            "config": Object {
              "configFolder": "apps/main-app/.storybook",
            },
            "outputPath": "dist/storybook/main-app",
            "projectBuildConfig": "main-app",
            "uiFramework": "@storybook/angular",
          },
          "outputs": Array [
            "{options.outputPath}",
          ],
        },
        "storybook": Object {
          "configurations": Object {
            "ci": Object {
              "quiet": true,
            },
          },
          "executor": "@nrwl/storybook:storybook",
          "options": Object {
            "config": Object {
              "configFolder": "apps/main-app/.storybook",
            },
            "port": 4400,
            "projectBuildConfig": "main-app",
            "uiFramework": "@storybook/angular",
          },
        },
      },
    },
    "ui-one": Object {
      "projectType": "library",
      "root": "libs/ui/one",
      "sourceRoot": "libs/ui/one/src",
      "targets": Object {
        "build-storybook": Object {
          "configurations": Object {
            "ci": Object {
              "quiet": true,
            },
          },
          "executor": "@nrwl/storybook:build",
          "options": Object {
            "config": Object {
              "configFolder": "libs/ui/one/.storybook",
            },
            "outputPath": "dist/storybook/ui/one",
            "projectBuildConfig": "ui-one:build-storybook",
            "uiFramework": "@storybook/angular",
          },
          "outputs": Array [
            "{options.outputPath}",
          ],
        },
        "storybook": Object {
          "executor": "@nrwl/storybook:storybook",
          "options": Object {
            "config": Object {
              "configFolder": "libs/ui/one/.storybook",
            },
            "port": 4400,
            "projectBuildConfig": "ui-one:build-storybook",
            "uiFramework": "@storybook/angular",
          },
        },
      },
    },
    "ui-three": Object {
      "projectType": "library",
      "root": "libs/ui/three",
      "sourceRoot": "libs/ui/three/src",
      "targets": Object {
        "build-storybook": Object {
          "configurations": Object {
            "ci": Object {
              "quiet": true,
            },
          },
          "executor": "@nrwl/storybook:build",
          "options": Object {
            "config": Object {
              "configFolder": "libs/ui/three/.storybook",
            },
            "outputPath": "dist/storybook/ui/three",
            "projectBuildConfig": "ui-three:build-storybook",
            "uiFramework": "@storybook/angular",
          },
          "outputs": Array [
            "{options.outputPath}",
          ],
        },
        "storybook": Object {
          "executor": "@nrwl/storybook:storybook",
          "options": Object {
            "config": Object {
              "configFolder": "libs/ui/three/.storybook",
            },
            "port": 4400,
            "projectBuildConfig": "ui-three:build-storybook",
            "uiFramework": "@storybook/angular",
          },
        },
      },
    },
    "ui-two": Object {
      "projectType": "library",
      "root": "libs/ui/two",
      "sourceRoot": "libs/ui/two/src",
      "targets": Object {
        "build-storybook": Object {
          "configurations": Object {
            "ci": Object {
              "quiet": true,
            },
          },
          "executor": "@nrwl/storybook:build",
          "options": Object {
            "config": Object {
              "configFolder": "libs/ui/two/.storybook",
            },
            "outputPath": "dist/storybook/ui/two",
            "projectBuildConfig": "ui-two:build-storybook",
            "uiFramework": "@storybook/angular",
          },
          "outputs": Array [
            "{options.outputPath}",
          ],
        },
        "storybook": Object {
          "executor": "@nrwl/storybook:storybook",
          "options": Object {
            "config": Object {
              "configFolder": "libs/ui/two/.storybook",
            },
            "port": 4400,
            "projectBuildConfig": "ui-two:build-storybook",
            "uiFramework": "@storybook/angular",
          },
        },
      },
    },
  },
  "version": 1,
}
`;

exports[`Set the projectBuildConfig option in the Storybook configuration for Angular projects for all types of angular projects - non-buildable and buildable libs/apps should still set the projectBuildConfig even if target names are not the default 1`] = `
Object {
  "projects": Object {
    "main-app": Object {
      "prefix": "katst",
      "projectType": "application",
      "root": "apps/main-app",
      "sourceRoot": "apps/main-app/src",
      "targets": Object {
        "lmfkcn": Object {
          "configurations": Object {
            "ci": Object {
              "quiet": true,
            },
          },
          "executor": "@nrwl/storybook:storybook",
          "options": Object {
            "config": Object {
              "configFolder": "apps/main-app/.storybook",
            },
            "port": 4400,
            "projectBuildConfig": "main-app",
            "uiFramework": "@storybook/angular",
          },
        },
        "njdfvndfjnv": Object {
          "executor": "@angular-devkit/build-angular:browser",
          "outputs": Array [
            "{options.outputPath}",
          ],
        },
        "odmwjbc": Object {
          "configurations": Object {
            "ci": Object {
              "quiet": true,
            },
          },
          "executor": "@nrwl/storybook:build",
          "options": Object {
            "config": Object {
              "configFolder": "apps/main-app/.storybook",
            },
            "outputPath": "dist/storybook/main-app",
            "projectBuildConfig": "main-app",
            "uiFramework": "@storybook/angular",
          },
          "outputs": Array [
            "{options.outputPath}",
          ],
        },
      },
    },
    "ui-one": Object {
      "projectType": "library",
      "root": "libs/ui/one",
      "sourceRoot": "libs/ui/one/src",
      "targets": Object {
        "asdgsdfg": Object {
          "configurations": Object {
            "ci": Object {
              "quiet": true,
            },
          },
          "executor": "@nrwl/storybook:build",
          "options": Object {
            "config": Object {
              "configFolder": "libs/ui/one/.storybook",
            },
            "outputPath": "dist/storybook/ui/one",
            "projectBuildConfig": "ui-one:asdgsdfg",
            "uiFramework": "@storybook/angular",
          },
          "outputs": Array [
            "{options.outputPath}",
          ],
        },
        "trthrngb": Object {
          "executor": "@nrwl/storybook:storybook",
          "options": Object {
            "config": Object {
              "configFolder": "libs/ui/one/.storybook",
            },
            "port": 4400,
            "projectBuildConfig": "ui-one:asdgsdfg",
            "uiFramework": "@storybook/angular",
          },
        },
      },
    },
    "ui-three": Object {
      "projectType": "library",
      "root": "libs/ui/three",
      "sourceRoot": "libs/ui/three/src",
      "targets": Object {
        "aaaa": Object {
          "configurations": Object {
            "ci": Object {
              "quiet": true,
            },
          },
          "executor": "@nrwl/storybook:build",
          "options": Object {
            "config": Object {
              "configFolder": "libs/ui/three/.storybook",
            },
            "outputPath": "dist/storybook/ui/three",
            "projectBuildConfig": "ui-three:aaaa",
            "uiFramework": "@storybook/angular",
          },
          "outputs": Array [
            "{options.outputPath}",
          ],
        },
        "nmkgd": Object {
          "executor": "@nrwl/storybook:storybook",
          "options": Object {
            "config": Object {
              "configFolder": "libs/ui/three/.storybook",
            },
            "port": 4400,
            "projectBuildConfig": "ui-three:aaaa",
            "uiFramework": "@storybook/angular",
          },
        },
      },
    },
    "ui-two": Object {
      "projectType": "library",
      "root": "libs/ui/two",
      "sourceRoot": "libs/ui/two/src",
      "targets": Object {
        "sdft": Object {
          "executor": "@nrwl/storybook:storybook",
          "options": Object {
            "config": Object {
              "configFolder": "libs/ui/two/.storybook",
            },
            "port": 4400,
            "projectBuildConfig": "ui-two:thjkkb",
            "uiFramework": "@storybook/angular",
          },
        },
        "thjkkb": Object {
          "configurations": Object {
            "ci": Object {
              "quiet": true,
            },
          },
          "executor": "@nrwl/storybook:build",
          "options": Object {
            "config": Object {
              "configFolder": "libs/ui/two/.storybook",
            },
            "outputPath": "dist/storybook/ui/two",
            "projectBuildConfig": "ui-two:thjkkb",
            "uiFramework": "@storybook/angular",
          },
          "outputs": Array [
            "{options.outputPath}",
          ],
        },
      },
    },
  },
  "version": 1,
}
`;

exports[`Set the projectBuildConfig option in the Storybook configuration for Angular projects for non-angular projects should not change their Storybook configuration 1`] = `
Object {
  "projects": Object {
    "main-app": Object {
      "prefix": "katst",
      "projectType": "application",
      "root": "apps/main-app",
      "sourceRoot": "apps/main-app/src",
      "targets": Object {
        "storybook": Object {
          "configurations": Object {
            "ci": Object {
              "quiet": true,
            },
          },
          "executor": "@nrwl/storybook:storybook",
          "options": Object {
            "config": Object {
              "configFolder": "apps/main-app/.storybook",
            },
            "port": 4400,
            "uiFramework": "@storybook/react",
          },
        },
      },
    },
    "ui-one": Object {
      "projectType": "library",
      "root": "libs/ui/one",
      "sourceRoot": "libs/ui/one/src",
      "targets": Object {
        "build-storybook": Object {
          "configurations": Object {
            "ci": Object {
              "quiet": true,
            },
          },
          "executor": "@nrwl/storybook:build",
          "options": Object {
            "config": Object {
              "configFolder": "libs/ui/one/.storybook",
            },
            "outputPath": "dist/storybook/ui/one",
            "uiFramework": "@storybook/angular",
          },
          "outputs": Array [
            "{options.outputPath}",
          ],
        },
        "storybook": Object {
          "executor": "@nrwl/storybook:storybook",
          "options": Object {
            "config": Object {
              "configFolder": "libs/ui/one/.storybook",
            },
            "port": 4400,
            "uiFramework": "@storybook/react",
          },
        },
      },
    },
  },
  "version": undefined,
}
`;
